sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -214,6 +228,10 @@ class StreamExecution(
       // While active, repeatedly attempt to run batches.
       SparkSession.setActiveSession(sparkSession)
 
+      updateStatusMessage("Initializing sources")
```

Comment:
Could you move `SparkSession.setActiveSession(sparkSession)`, `updateStatusMessage("Initializing sources")` and `logicalPlan` above `postEvent(new QueryStartedEvent(id, runId, name))`? Then we can initialize the logical plan before returning `StreamingQuery` to the user and ensure `logicalPlan` is always initialized in the stream thread. And you can also add `assert(Thread.currentThread().isInstanceOf[StreamExecutionThread])` when creating `logicalPlan`, and remove the unnecessary test you added since this assertion will be tested in all current stream tests.